disnake/ui/button.py

```diff
@@ -46,7 +65,14 @@
     from .view import View
 
 B = TypeVar("B", bound="Button")
+B_co = TypeVar("B_co", bound="Button", covariant=True)
 V = TypeVar("V", bound="Optional[View]", covariant=True)
+P = ParamSpec("P")
+
+
+class ButtonObject(Protocol[B_co, P]):
```

Comment:
Looks like `B_co` is unused here? Same thing in `SelectObject` below, but maybe I'm just not seeing its purpose

Reply:
Uh-oh, true. It's purpose for the decorator is checking the cls type against the type of the callback's second argument. Forgot to make it meaningful for the Protocol itself though